creed/tiling_utils_test.py

```diff
@@ -0,0 +1,745 @@
+            actual_center_points_sorted.insert(mi, (14540, -10830))
+
+        # easiest case: the center points should be sorted
+        if randomize_setting == ['N', 'N']:
+            assert center_points == actual_center_points_sorted
+        # if only the second fov is randomized
+        elif randomize_setting == ['N', 'Y']:
+            # ensure the fov 1 center points are the same for both sorted and random
+            assert center_points[:fov_1_end_pos] == actual_center_points_sorted[:fov_1_end_pos]
+
+            # ensure the random center points for fov 2 contain the same elements
+            # as its sorted version
+            misc_utils.verify_same_elements(
+                computed_center_points=center_points[fov_1_end_pos:],
+                actual_center_points=actual_center_points_sorted[fov_1_end_pos:]
+            )
+
+            # however, fov 2 sorted entries should NOT equal fov 2 random entries
+            # NOTE: due to randomization, this test will fail once in a blue moon
+            assert center_points[fov_1_end_pos:] != actual_center_points_sorted[fov_1_end_pos:]
+        # if both fovs are randomized
+        elif randomize_setting == ['Y', 'Y']:
+            # ensure the random center points for fov 1 contain the same elements
+            # as its sorted version
+            misc_utils.verify_same_elements(
+                computed_center_points=center_points[:fov_1_end_pos],
+                actual_center_points=actual_center_points_sorted[:fov_1_end_pos]
+            )
+
+            # however, fov 1 sorted entries should NOT equal fov 1 random entries
+            # NOTE: due to randomization, this test will fail once in a blue moon
+            assert center_points[:fov_1_end_pos] != actual_center_points_sorted[:fov_1_end_pos]
+
+            # ensure the random center points for fov 2 contain the same elements
+            # as its sorted version
+            misc_utils.verify_same_elements(
+                computed_center_points=center_points[fov_1_end_pos:],
+                actual_center_points=actual_center_points_sorted[fov_1_end_pos:]
+            )
+
+            # however, fov 2 sorted entries should NOT equal fov 2 random entries
+            # NOTE: due to randomization, this test will fail once in a blue moon
+            assert center_points[fov_1_end_pos:] != actual_center_points_sorted[fov_1_end_pos:]
+
+
+@parametrize_with_cases('top_left, top_right, bottom_left, bottom_right',
+                        cases=test_cases.ValidateRhombusCoordsTests, glob='*_failure')
+def test_validate_tma_corners_failure(top_left, top_right, bottom_left, bottom_right):
+    # error checks 1-4: test invalid top_left, top_right, bottom_left, bottom_right respectively
+    with pytest.raises(ValueError):
+        tiling_utils.validate_tma_corners(top_left, top_right, bottom_left, bottom_right)
+
+
+@parametrize_with_cases('top_left, top_right, bottom_left, bottom_right',
```

Comment:
You can mark the cases that are expected to fail with `xfail`; see the rosetta PR. This will allow you to combine these functions together